tests/dialects/test_arith.py

```diff
@@ -49,7 +49,7 @@
 from xdsl.ir import Attribute
 from xdsl.utils.exceptions import VerifyException
 
-_BinOpT = TypeVar("_BinOpT", bound=BinaryOperation)
+_BinOpT = TypeVar("_BinOpT", bound=Attribute)
```

Comment:
why this change?

Reply:
This is basically Pyright shennanigans in Pytest parameterized test. The test is parameterized by the Operation class (Addi, Subi, ...). It used to be happy with `type[_BinOpT]` (I would argue it was overkill and `type[BinaryOperation]` should have yielded the same thing, but I didn't play around)
Now that it's generic, fields and such are actually defined by `BinaryOperation`, so it makes Pyright happier. But it requires a type parameter! So I switch to `type[BinaryOperation[_binOpT]]`